pkg/registry/apis/folders/register.go

```diff
@@ -246,6 +247,15 @@ func (b *FolderAPIBuilder) Validate(ctx context.Context, a admission.Attributes,
 
 	obj := a.GetObject()
 
+	f, ok := obj.(*v0alpha1.Folder)
```

Comment:
🤔 I wonder if this should go before the for loop, you're right that we should double check if we are dealing with a folder first things first

Reply:
It could be, but I think it makes no difference, as the validation above is based on the `name` attribute, which is already provided by the `a.GetName()` method